client/src/includes/a11y-result.ts

```diff
@@ -70,6 +72,101 @@ export const getAxeConfiguration = (
   return null;
 };
 
+/**
+ * Custom rule for checking image alt text. This rule checks if the alt text for images
+ * contains file extensions or URLs, which are considered poor quality alt text.
+ * The rule will be added via the Axe.configure() API.
+ * https://github.com/dequelabs/axe-core/blob/master/doc/API.md#api-name-axeconfigure
+ */
+export const checkImageAltText = (node: Element) => {
+  const imageFileExtensions = [
+    'apng',
+    'avif',
+    'gif',
+    'jpg',
+    'jpeg',
+    'jfif',
+    'pjpeg',
+    'pjp',
+    'png',
+    'svg',
+    'tif',
+    'webp',
+  ];
+  const imageURLs = ['http://', 'https://', 'www.'];
+  const altTextAntipatterns = new RegExp(
+    `\\.(${imageFileExtensions.join('|')})|(${imageURLs.join('|')})`,
+    'i',
+  );
+
+  const image = node as HTMLImageElement;
+  const altText = image.getAttribute('alt') || '';
+
+  const hasBadAltText = altTextAntipatterns.test(altText);
+  return !hasBadAltText;
+};
+
+/**
+ * Defines custom Axe rules, mapping each check to its corresponding JavaScript function.
+ * This object holds the custom checks that will be added to the Axe configuration.
+ */
+const customChecks = {
+  'check-image-alt-text': checkImageAltText,
+  // Add other custom checks here
+};
+
+/**
+ * Configures custom Axe rules by integrating the custom checks with their corresponding
+ * JavaScript functions. It modifies the provided configuration to include these checks.
+ */
+const addCustomChecks = (customConfig: Spec): Spec => {
```

Comment:
I think this is worth writing unit tests for. And I’d prefer the parameter name was `spec` if it’s of this `Spec` type.